project/profiles/capacity4more/modules/c4m/content/c4m_content_group/c4m_content_group.module

```diff
@@ -915,3 +916,165 @@ function c4m_content_group_node_view($node, $view_mode, $langcode) {
+  $commands[] = ajax_command_replace('#toggle-group-notifications', $link);
+  $page = array('#type' => 'ajax', '#commands' => $commands);
+  ajax_deliver($page);
+}
+
+/**
+ * Creates html for link to toggle group notifications.
+ *
+ * @param object $account
+ *   User, for which link is created.
+ * @param object $group
+ *   Group, for which link is created.
+ *
+ * @return Html
+ *    Html block representing the link.
+ */
+function c4m_content_group_create_toggle_notification_link($account, $group) {
+  $user_subscribed = _c4m_content_group_is_subscribed_to_group_notifications($account, $group);
+  $message_text = $user_subscribed ? t('Notifications are enabled') : t('Notifications are disabled');
+  $link_text = $user_subscribed ? t('disable') : t('enable');
+
+  $options = array(
+    'attributes' => array(
+      'class' => 'use-ajax',
+    ),
+    'purl' => array('disabled' => TRUE),
+    'html' => TRUE,
+  );
+
+  $link = l($link_text, url('group/' . $group->nid . '/notification/nojs'), $options);
+
+  $tag['element'] = array(
+    '#tag' => 'div',
+    '#value' => $message_text . '  (' . $link . ')',
+    '#attributes' => array(
+      'id' => 'toggle-group-notifications',
+    ),
+  );
+
+  return theme_html_tag($tag);
+}
+
+/**
+ * Determines if user is subscribed to group notifications, or not.
+ *
+ * @param object $account
+ *   Checked user.
+ * @param object $group
+ *   Group for which notifications are checked.
+ *
+ * @return bool
+ *   Whether user is subscribed to group notifications, or not.
+ */
+function _c4m_content_group_is_subscribed_to_group_notifications($account, $group) {
+  $user_flags = flag_get_user_flags('node', $group->nid);
+  $user_subscribed = array_key_exists('subscribe_c4m_follow_group', $user_flags);
+
+  if ($user_subscribed) {
+    $wrapper = entity_metadata_wrapper('user', $account);
+    foreach ($wrapper->c4m_optout_group->value(array('identifier' => TRUE)) as $existing_id) {
+      if ($existing_id == $group->nid) {
+        // User unsubscribed from entity.
+        $user_subscribed = FALSE;
+        break;
+      }
+    }
+  }
+
+  return $user_subscribed;
+}
+
+/**
+ * Enables / Disables group notifications for group member.
+ *
+ * @param object $account
+ *   Group member user.
+ * @param object $group
+ *   Group for which notifications are toggled.
+ */
+function _c4m_content_group_toggle_group_notifications($account, $group) {
+  $user_subscribed = _c4m_content_group_is_subscribed_to_group_notifications($account, $group);
+
+  if ($user_subscribed) {
+    // Unflag.
+    flag('unflag', 'subscribe_c4m_follow_group', $group->nid);
+  }
+  else {
+    // Flag.
+    flag('flag', 'subscribe_c4m_follow_group', $group->nid);
+    // Unset optout field.
+    $wrapper = entity_metadata_wrapper('user', $account);
+    $unsubscribed = $wrapper->c4m_optout_group->value(array('identifier' => TRUE));
+
+    if (($index = array_search($group->nid, $unsubscribed)) !== FALSE) {
```

Comment:
@ordavidil 
The purpose of this `if` is to determine if needle exists at array, or not.
array_search() returns the key for needle if it is found in the array, `FALSE` otherwise.
If needle is found at key = 0, `if (array_search())` is translated to `if (0)`, and we get negative answer, while we should get a positive one.
Therefore, when checking if needle is found at array or not, we must compare to `FALSE`.